client/src/components/TaskOptionsModal.jsx

```diff
@@ -54,24 +54,50 @@ start: "2022-04-15T09:30:20"
 title: "meeting with A"
 
 var TaskOptionsModal = (props) => {
+  const [userTask, setUserTask] = useState(props.task)
   const [todoTitle, setTodoTitle] = useState(props.task.title);
   const [todoDescription, setTodoDescription] = useState(props.task.description);
-  const [startTime, setStartTime] = useState(props.task.start);
-  const [endTime, setEndTime] = useState(props.task.end_date);
+  const [startTime, setStartTime] = useState(userTask.start);
+  const [endTime, setEndTime] = useState(userTask.end_date);
 
   const classes = useStyles();
 
   const handleEditDone = () => {
+    //front end update
+    const taskCopy = userTask
+    taskCopy.title = todoTitle;
+    taskCopy.description = todoDescription;
+    taskCopy.start = startTime;
+    taskCopy.end_date = endTime
+    let hours = endTime.getHours() - startTime.getHours()
+    let minutes = endTime.getMinutes() - startTime.getMinutes()
+    if (minutes < 0) {
+      const convertedHours = (hours * 60) + minutes;
+      hours = Math.floor(convertedHours/60)
+      minutes = convertedHours % 60
+
+    }
+    const duration = hours + ':' + minutes
+    taskCopy.duration = duration
+    setUserTask(taskCopy)
+    props.updateTask(taskCopy)
```

Comment:
Awesome. This looks like a good start to the duration part. Def will need to use this Modal now for the "Add" button.